dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/security/DefaultSecurityService.java

```diff
@@ -144,6 +117,21 @@ public void setI18nManager( I18nManager i18nManager )
 
     @Autowired
     private RestTemplate restTemplate;
+    
+    @Autowired
+    private CacheProvider cacheProvider;
+    
+    // -------------------------------------------------------------------------
+    // Initialization
+    // -------------------------------------------------------------------------
+
+    
+    @PostConstruct
+    public void init()
+    {
+        this.userFailedLoginAttemptCache = cacheProvider.createCacheInstance( "userFailedLoginAttempt", false, LOGIN_LOCKOUT_MINS*SECONDS_PER_MIN,  -1 , 0 );
+        
```

Comment:
I suggest some sort of builder pattern for initializing the caches. We might want to add additional properties later and we don't want many overloaded methods. Something along these lines:

cacheProvider.getCache( CacheParams
    .withKey( "userFailedLoginAttempt" )
    .withXxx( false )
    .withExpiry( s ).build() );